test/shallow_ice_test.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2017-2019 by Daniel Shapero <shapero@uw.edu>
+#
+# This file is part of icepack.
+#
+# icepack is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# The full text of the license can be found in the file LICENSE in the
+# icepack source directory or at <http://www.gnu.org/licenses/>.
+
+import numpy as np
+
+# Test our numerical solvers against this analytical solution.
+import pygmsh
+import os
+import matplotlib.pyplot as plt
+import firedrake
+from firedrake import inner, grad, dx, interpolate, as_vector, max_value
+import icepack, icepack.models
+from icepack.constants import (ice_density as ρ_I, water_density as ρ_W,
+                               glen_flow_law as n, weertman_sliding_law as m,
+                               gravity as g)
+
+R = 100e3
+alpha = firedrake.Constant(R)
+T = 254.15
+A = icepack.rate_factor(T)
+A0 = 2 * A * (ρ_I * g) ** n / (n + 2)
+
+def make_mesh(R, δx):
+    geometry = pygmsh.built_in.Geometry()
+    x1 = geometry.add_point([-R, 0, 0], lcar=δx)
+    x2 = geometry.add_point([+R, 0, 0], lcar=δx)
+    center1 = geometry.add_point([0, 0, 0,], lcar=δx)
+    arcs = [geometry.add_circle_arc(x1, center1, x2),
+            geometry.add_circle_arc(x2, center1, x1)]
+    line_loop = geometry.add_line_loop(arcs)
+    plane_surface = geometry.add_plane_surface(line_loop)
+    physical_lines = [geometry.add_physical(arc) for arc in arcs]
+    physical_surface = geometry.add_physical(plane_surface)
+    with open('shallow-ice.geo', 'w') as geo_file:
+        geo_file.write(geometry.get_code())    
+    transform_file = "gmsh -v 0 -2 -format msh2 -o shallow-ice.msh shallow-ice.geo"
+    os.system(transform_file)
+    mesh = firedrake.Mesh('shallow-ice.msh')
+    remove_geo = "rm shallow-ice.geo"
+    remove_mesh = "rm shallow-ice.msh"
+    os.system(remove_geo)
+    os.system(remove_mesh)
+    return mesh
+
+
+# Pick a geometry in order to have an exact solution. We'll use the
+# Bueler profile.
```

Comment:
If people are reading the comments they might not know offhand what the Bueler profile is, so a reference to the right chapter and section of Greve and Blatter could be helpful.